Emit events using Borsh encoding for Solana (#1008)
hyperledger/solang: docs/language/events.rst

fields are stored in the data section of the event. The event name does not need to be unique; just like
functions, they can be overloaded as long as the fields are of different types, or the event has
a different number of arguments.

.. warning::

>     On Solana, writing ``indexed`` besides an event field has no impact when emitting events. The ``indexed``
>     keyword serves only to generate metadata information for IDL files. All event attributes will be encoded as data to
>     be passed for Solana's ``sol_log_data`` system call, regardless of the ``indexed`` keyword being present. This
>     behavior follows what Solana's Anchor framework does.

Reviewer: ```suggestion
    On Solana, the ``indexed`` event field attribute has no effect. All event attributes will be encoded as data to
    be passed for Solana's ``sol_log_data`` system call, regardless if the ``indexed`` keyword is present or not.
    This behavior follows what Solana's Anchor framework does.
```
